src/components/Files/FileFolderSelect.tsx

```diff
@@ -0,0 +1,278 @@
+import {
+  FloatingFocusManager,
+  FloatingOverlay,
+  FloatingPortal,
+  useDismiss,
+  useFloating,
+  useId,
+  useInteractions,
+  useRole,
+} from '@floating-ui/react';
+import { IconCaretRightFilled, IconX } from '@tabler/icons-react';
+import { ChangeEvent, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import { getChildAndCurrentFoldersIdsById } from '@/src/utils/app/folders';
+
+import { FilesActions, FilesSelectors } from '@/src/store/files/files.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import { UIActions } from '@/src/store/ui/ui.reducers';
+
+import FolderPlus from '../../../public/images/icons/folder-plus.svg';
+import Folder from '../Folder';
+import Spinner from '../Spinner';
+
+interface Props {
+  isOpen: boolean;
+  selectedFolderName: string | undefined;
+  onClose: (path: string | undefined | boolean) => void;
+}
+
+export const FileFolderSelect = ({
+  isOpen,
+  selectedFolderName,
+  onClose,
+}: Props) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation('chat');
+
+  const { refs, context } = useFloating({
+    open: isOpen,
+    onOpenChange: onClose,
+  });
+  const role = useRole(context);
+  const dismiss = useDismiss(context, { outsidePressEvent: 'mousedown' });
+  const { getFloatingProps } = useInteractions([role, dismiss]);
+  const headingId = useId();
+
+  const folders = useAppSelector(FilesSelectors.selectFolders);
+  const [searchQuery, setSearchQuery] = useState('');
+  const filteredFolders = useMemo(() => {
+    return folders.filter(({ name }) =>
+      name.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+  }, [folders, searchQuery]);
+  const foldersStatus = useAppSelector(FilesSelectors.selectFoldersStatus);
+  const loadingFolderId = useAppSelector(FilesSelectors.selectLoadingFolderId);
+  const newAddedFolderId = useAppSelector(
+    FilesSelectors.selectNewAddedFolderId,
+  );
+  const [openedFoldersIds, setOpenedFoldersIds] = useState<string[]>([]);
+  const [isAllFilesOpened, setIsAllFilesOpened] = useState(true);
+
+  const [selectedFolderId, setSelectedFolderId] = useState<string | undefined>(
+    selectedFolderName,
+  );
+
+  useEffect(() => {
+    if (isOpen) {
+      dispatch(FilesActions.getFolders({}));
+      openedFoldersIds.forEach((folderId) => {
+        dispatch(FilesActions.getFolders({ path: folderId }));
+      });
```

Comment:
executing `dispatch` in cycle isn't good practice.
Add some action which can work with array of ids:
```suggestion
      dispatch(FilesActions.getFolderList(openedFoldersIds));
```
and what does `dispatch(FilesActions.getFolders({}));` do? Can it be included into `getFolderList`? e.g. with flag `uploadRoot`